components/layout_2020/positioned.rs

```diff
@@ -928,33 +943,69 @@ impl<'a> AbsoluteAxisSolver<'a> {
             _ => return None,
         };
 
+        assert_eq!(
+            container_writing_mode.is_horizontal(),
+            abspos_writing_mode.is_horizontal(),
+            "Mixed horizontal and vertical writing modes are not supported yet"
+        );
+
         let mut value_after_safety = self.alignment.value();
         if self.alignment.flags() == AlignFlags::SAFE &&
             margin_box_axis.length > alignment_container.length
         {
             value_after_safety = AlignFlags::START;
         }
 
-        match value_after_safety {
-            AlignFlags::CENTER | AlignFlags::SPACE_AROUND | AlignFlags::SPACE_EVENLY => Some(
-                alignment_container.origin +
-                    ((alignment_container.length - margin_box_axis.length) / 2),
-            ),
-            AlignFlags::FLEX_END | AlignFlags::END => Some(
-                alignment_container.origin + alignment_container.length - margin_box_axis.length,
-            ),
-            _ => None,
-        }
+        let self_value_matches_container = self.axis == AxisDirection::Block ||
+            container_writing_mode.is_bidi_ltr() == abspos_writing_mode.is_bidi_ltr();
```

Comment:
I suspect we actually need to compare with `shared_fragment.original_parent_writing_mode` instead of `container_writing_mode` when both offsets are auto. And rather than passing 3 writing modes around, it may be better to store them in the `AbsoluteAxisSolver`.

But not sure, I think it's better to land #34426 first and then we can test.

Reply:
This is the relevant testcase:

```html
<!DOCTYPE html>
<div style="position: relative">
  <div style="display: flex; flex-direction: column; width: 100px; height: 100px; border: solid; direction: rtl">
    <div style="position: absolute; width: 20px; height: 20px; background: cyan; top: 20px; align-self: self-start"></div>
    <div style="position: absolute; width: 20px; height: 20px; background: magenta; bottom: 20px; align-self: self-end"></div>
  </div>
</div>
```

I have rebased your patch and pushed a fix.